Rewrite to use new API

ErikBoesen opened the pull request:
This pull request contains a number of changes I've made in recent months, including updating to use YaleDine's API, adding a date selector to the dining hall page, rebuilding the internal SQLite data management system to use Room, and completing the app's rebranding to Yale Menus.

ErikBoesen commented on app/src/main/java/com/adisa/diningplus/adapters/MainListAdapter.java:
```diff
@@ -42,32 +45,32 @@ public int compare(MainActivity.HallItem o1, MainActivity.HallItem o2) {
+        shieldMap.put("SY", R.drawable.saybrook);
+        shieldMap.put("SM", R.drawable.silliman);
+        shieldMap.put("TD", R.drawable.td);
+        shieldMap.put("TC", R.drawable.trumbull);
```
You'll notice I've switched to using location codes instead of names. I decided to do this because names can be ambiguous. For example, in my new YaleDine API, I override names to be the most verbose variation for `name`, for example Ezra Stiles or Benjamin Franklin, and give any possible shorter name (Stiles, Franklin, JE) in the `shortname` property. Thus I thought using abbreviations (which are now served by my API in the `code` property) would be more proper.

ErikBoesen commented on app/src/main/java/com/adisa/diningplus/adapters/MainListAdapter.java:
```diff
@@ -42,32 +45,32 @@ public int compare(MainActivity.HallItem o1, MainActivity.HallItem o2) {
-        Collections.sort(closedHalls, hallSort);
-        halls.addAll(openHalls);
-        halls.addAll(closedHalls);
+        locations = new ArrayList<>();
```
I renamed "halls" to "locations" to match YaleDine's format and be more generic for other types of dining location.

amalik12 reviewed (commented):
Also, it looks like the build is failing.

ErikBoesen commented on app/src/main/java/com/adisa/diningplus/activities/LocationActivity.java:
```diff
@@ -0,0 +1,286 @@
+        }
+    }
+
+    public void previousDate(View view) {
+        this.date.add(Calendar.DATE, -1);
+        this.getMeals();
+    }
+
+    public void nextDate(View view) {
+        this.date.add(Calendar.DATE, 1);
+        this.getMeals();
+    }
+
+    public void chooseDate(View view) {
+    }
```
Not yet, but they will be. Hopefully today I'm going to make the date changing buttons work.

amalik12 commented:
The allergen popup when the app is first opened doesn't seem to be saving selections. Also, all of the menus show up as "No meals available" in the app, is this expected?

ErikBoesen commented:
@amalik12 I've fixed the allergen popup. The empty lists are to be expected since nothing is being served right now, but I've added dummy data for Hopper today (Jan 21).

amalik12 commented:
The allergen popup works now! I'm not seeing any data for Hopper when I change the date to the 21st and the system time to 8am, though. Also, the room favoriting feature doesn't seem to be working.

ErikBoesen commented:
Update: app is now mostly converted to Kotlin.

ErikBoesen commented:
Alright, this rebuild is basically done now. Let me know if you find any other issues.

amalik12 commented:
This gets thrown when I try to load the dining hall list:
```W/System.err: org.json.JSONException: No value for shortname
        at org.json.JSONObject.get(JSONObject.java:399)
        at org.json.JSONObject.getString(JSONObject.java:560)
        at com.adisa.diningplus.db.entities.Hall.fromJSON(Hall.java:30)
        at com.adisa.diningplus.network.API.getHalls(API.kt:49)
        at com.adisa.diningplus.activities.MainActivity$UpdateTask.doInBackground(MainActivity.java:264)
        at com.adisa.diningplus.activities.MainActivity$UpdateTask.doInBackground(MainActivity.java:253)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
```

ErikBoesen commented:
@amalik12 Tests should pass now.

amalik12 reviewed (approved):
I'm able to view the menus now, this looks great! I found a few minor bugs, I'll open issues for them and you can tackle them in separate PRs.
